File: CivitasJava/src/civitas/SorpresaPorCasaHotel.java
/*
 * To change this license header, choose License Headers in Project Properties.
 * To change this template file, choose Tools | Templates
 * and open the template in the editor.
 */
package civitas;

import java.util.ArrayList;

/**
 *
 * @author luis
 */
public class SorpresaPorCasaHotel extends Sorpresa{
    private int valor;
    
    SorpresaPorCasaHotel(int valor){
        super("SorpresaPorCasaHotel");
        this.valor = valor;
    }
    
    @Override
    public void aplicarAJugador(int actual, ArrayList<Jugador> todos){
        if(jugadorCorrecto(actual,todos)){
            informe(actual,todos);
            int casasHoteles = 0;
            
            for(int i = 0; i < todos.get(actual).getPropiedades().size(); i++){
                casasHoteles = casasHoteles + todos.get(actual).getPropiedades().get(i).cantidadCasasHoteles();
            }
            
            todos.get(actual).modificarSaldo(this.valor*casasHoteles);
        }
    }
    
    @Override
    public String toString(){
        String e = "\nNombre: SorpresaPorCasaHotel";
        return e;
    }
}

File: CivitasRuby/lib/sorpresa_por_casa_hotel.rb
# encoding:utf-8
# To change this license header, choose License Headers in Project Properties.
# To change this template file, choose Tools | Templates
# and open the template in the editor.

module Civitas
  class SorpresaPorCasaHotel < Sorpresa
    
    def initialize(valor)
      super("SorpresaPorCasaHotel")
      @valor = valor
    end
    
    def aplicar_a_jugador(actual, todos)
      if jugador_correcto(actual,todos)
        informe(actual,todos)
        todos[actual].modificar_saldo(todos[actual].cantidad_casa_hoteles*@valor)
      end
    end
    
        
    def to_s
      "\nNombre: #{@texto}"
    end
    
       
    public_class_method  :new
    
  end
end
